sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -47,6 +46,24 @@ def test_partial_dependence_classifier():
     assert axes is None
     assert_array_equal(pdp, pdp_2)
 
+    # with trivial (no-op) sample weights
+    clf.fit(X, y, sample_weight=np.ones(len(y)))
+
+    pdp_w, axes_w = partial_dependence(clf, [0], X=X, grid_resolution=5)
+
+    assert pdp_w.shape == (1, 4)
+    assert axes_w[0].shape[0] == 4
+    assert_allclose(pdp_w, pdp)
+
+    # with non-trivial sample weights
+    clf.fit(X, y, sample_weight=sample_weight)
+
+    pdp_w2, axes_w2 = partial_dependence(clf, [0], X=X, grid_resolution=5)
+
+    assert pdp_w2.shape == (1, 4)
+    assert axes_w2[0].shape[0] == 4
+    assert np.sum(np.abs(pdp_w2 - pdp_w)) / np.sum(np.abs(pdp_w)) > 0.05
```

Comment:
Can you remove the sum to check element-wise? If not, can you use `mean` instead so that the threshold isn't dependent on the size